### PR TITLE
[netcore] Implement AssemblyLoadContext.GetLoadedAssemblies()

### LINKED ISSUE #14855: [netcore]Make Microsoft.VisualBasic.ApplicationServices.Tests.AssemblyInfoTests.LoadedAssemblies pass
The current test fails with:

```
   Microsoft.VisualBasic.ApplicationServices.Tests.AssemblyInfoTests.LoadedAssemblies [FAIL]
      System.NotImplementedException : The method or operation is not implemented.
      Stack Trace:
           at Microsoft.VisualBasic.ApplicationServices.AssemblyInfo.get_LoadedAssemblies()
           at Microsoft.VisualBasic.ApplicationServices.Tests.AssemblyInfoTests.LoadedAssemblies()
           at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
  Finished:    Microsoft.VisualBasic.Core.Tests
```

Determine why this is failing and try to make it pass.

### PR DESCRIPTION
**Pull request by CoffeeFlux**
Fixes #14855

### CODE REVIEW

File: mono/metadata/appdomain.c [hunk 1]
```diff
@@ -1324,6 +1302,24 @@ ves_icall_System_AppDomain_GetAssemblies (MonoAppDomainHandle ad, MonoBoolean re
 	return res;
 }
 
+MonoArrayHandle
+ves_icall_System_AppDomain_GetAssemblies (MonoAppDomainHandle ad, MonoBoolean refonly, MonoError *error)
```
Comment (lambdageek) [inline mono/metadata/appdomain.c]: Do we need this one in netcore? can we ifdef it out?

File: mono/metadata/appdomain.c [hunk 2]
```diff
@@ -1324,6 +1302,24 @@ ves_icall_System_AppDomain_GetAssemblies (MonoAppDomainHandle ad, MonoBoolean re
...
+	MonoDomain *domain = MONO_HANDLE_GETVAL (ad, data);
+	return get_assembly_array_from_domain (domain, refonly, error);
+}
+
+#ifdef ENABLE_NETCORE
+MonoArrayHandle
+ves_icall_System_Runtime_Loader_AssemblyLoadContext_InternalGetLoadedAssemblies (MonoBoolean refonly, MonoError *error)
+{
+	error_init (error);
+	MonoDomain *domain = mono_domain_get ();
```
Comment (vargaz) [inline mono/metadata/appdomain.c]: No need to init the error.

Comment (vargaz) [reply mono/metadata/appdomain.c]: I think netcore has no refonly assemblies.

File: mono/metadata/appdomain.c [hunk 3]
```diff
@@ -1324,6 +1302,24 @@ ves_icall_System_AppDomain_GetAssemblies (MonoAppDomainHandle ad, MonoBoolean re
...
+	return get_assembly_array_from_domain (domain, refonly, error);
```
Comment (lambdageek) [inline mono/metadata/appdomain.c]: I think we can specialize to `refonly == FALSE` for netcore. looks like `ReflectionOnlyGetAssemblies` just always returns an empty array.
https://github.com/mono/mono/blob/090dffb9a09f26dc250d8d09a46819d192290b0b/netcore/System.Private.CoreLib/shared/System/AppDomain.cs#L183

Comment (CoffeeFlux) [reply mono/metadata/appdomain.c]: I left the parameter and just pass false, but I can remove it entirely from the icall.

File: mono/metadata/appdomain.c (hunk 2 above)
Comment (CoffeeFlux) [reply mono/metadata/appdomain.c]: Do I not need to init the error for both this and the AppDomain call, or just this one? I'm not sure I understand why it's not needed here.

File: mono/metadata/appdomain.c (hunk 3 above)
Comment (lambdageek) [reply mono/metadata/appdomain.c]: I'd just get rid of it entirely.

File: mono/metadata/appdomain.c (hunk 2 above)
Comment (lambdageek) [reply mono/metadata/appdomain.c]: You shouldn't need it in both.

We changed how MonoError works at one point but didn't finish updating the code.  used to be that you had to call `error_init (error)` on entry to any function with a `MonoError *error` argument.

These days you no longer need to do that.  Instead the initialization is done once when the error is declared using `ERROR_DECL (error)`. In icalls it's done with macros when we make the handles wrapper https://github.com/mono/mono/blob/090dffb9a09f26dc250d8d09a46819d192290b0b/mono/metadata/icall-table.h#L482

File: mono/metadata/icall-def.h [hunk 4]
```diff
@@ -177,7 +177,9 @@ HANDLES(KPAIR_5, "_ProtectUser", ves_icall_Mono_Security_Cryptography_KeyPairPer
 ICALL_TYPE(APPDOM, "System.AppDomain", APPDOM_23)
 HANDLES(APPDOM_23, "DoUnhandledException", ves_icall_System_AppDomain_DoUnhandledException, void, 2, (MonoAppDomain, MonoException))
 HANDLES(APPDOM_1, "ExecuteAssembly", ves_icall_System_AppDomain_ExecuteAssembly, gint32, 3, (MonoAppDomain, MonoReflectionAssembly, MonoArray))
+#ifndef ENABLE_NETCORE
```
Comment (lambdageek) [inline mono/metadata/icall-def.h]: Not needed.  This whole file is inside an ifdef.

Comment (CoffeeFlux) [reply mono/metadata/icall-def.h]: Oh right. Oops.

**Comment by CoffeeFlux** [comment]
@monojenkins build Linux x64 Bitcode

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

**Pull request merged**